.github/workflows/deploy_code.yaml

```diff
@@ -20,24 +20,22 @@ env:
 
 jobs:
   get_functions_and_tasks:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     name: Get functions and tasks
+    outputs:
+      all_functions: ${{ steps.get_list_of_functions.outputs.all_functions }}
+      all_ecs_tasks: ${{ steps.get_list_of_ecs_tasks.outputs.all_ecs_tasks }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
       - name: Get list of functions
         id: get_list_of_functions
-        run: echo "all_functions=\"$(cd ./src/functions && ls -d */ | sed 's#/##' | jq -R | jq -sc | sed 's/"/\\"/g')\"" >> "$GITHUB_OUTPUT"
+        run: echo "all_functions=$(cd ./src/functions && ls -d */ | sed 's#/##' | jq -R | jq -sc)" >> "$GITHUB_OUTPUT"
 
       - name: Get list of ECS tasks
         id: get_list_of_ecs_tasks
-        run: echo "all_ecs_tasks=\"$(cd ./src/tasks && ls -d */ | sed 's#/##' | jq -R | jq -sc | sed 's/"/\\"/g')\"" >> "$GITHUB_OUTPUT"
-
-      - name: echo outputs
-        run: |
-          echo "all_functions=${{ steps.get_list_of_functions.outputs.all_functions }}"
-          echo "all_ecs_tasks=${{ steps.get_list_of_ecs_tasks.outputs.all_ecs_tasks }}"
+        run: echo "all_ecs_tasks=$(cd ./src/tasks && ls -d */ | sed 's#/##' | jq -R | jq -sc)" >> "$GITHUB_OUTPUT"
 
   check_for_function_changes:
     uses: ./.github/workflows/get_directories_with_changes.yaml
```

Comment:
Think if check can be added back here to only run this job if we're not deploying all functions